ShiftLogger.Bata13/STUDY.ASP.ShiftLoggerTryThreeUserInterface/ProgramEngine.cs

```diff
@@ -0,0 +1,200 @@
+﻿using Spectre.Console;
+using STUDY.ASP.ShiftLoggerTryThree.Models;
+using System.Net.Http.Json;
+
+namespace STUDY.ASP.ShiftLoggerTryThreeUserInterface;
+internal class ProgramEngine
+{
+    public static void ViewAllShiftLogs(HttpClient client, string ApiBaseUrl)
+    {
+        var shiftLogs = client.GetFromJsonAsync<ShiftLogger[]>(ApiBaseUrl).Result;
+        if (shiftLogs != null)
+        {
+            var table = new Table();
+
+            table.AddColumn(new TableColumn("Id").Centered());
+            table.AddColumn(new TableColumn("First Name").Centered());
+            table.AddColumn(new TableColumn("Last Name").Centered());
+            table.AddColumn(new TableColumn("Clock In").Centered());
+            table.AddColumn(new TableColumn("Clock Out").Centered());
+            table.AddColumn(new TableColumn("Duration").Centered());
+
+            foreach (var shiftLog in shiftLogs)
+            {
+                TimeSpan duration = shiftLog.ClockOut - shiftLog.ClockIn;                
+                duration = TimeSpan.FromMinutes(Math.Round(duration.TotalMinutes));
+                string durationString = $"{(int)duration.TotalHours}h {duration.Minutes}m";
+
+                table.AddRow(
+                    $"{shiftLog.Id}",
+                    $"{shiftLog.EmployeeFirstName}",
+                    $"{shiftLog.EmployeeLastName}",
+                    $"{shiftLog.ClockIn}",
+                    $"{shiftLog.ClockOut}",
+                    $"[red]{durationString}[/]"
+                );
+            }
+
+            AnsiConsole.Write(table);
```

Comment:
From line 13 to 38 could be separated for example to `ConsoleEngine.ShowTable(List<ShiftLogs> shifts)` static method, because you use it 2 times which would save you about 25 lines of code.